docs/visual-basic/misc/bc42328.md

```diff
@@ -10,12 +10,102 @@ helpviewer_keywords:
 ms.assetid: 1561e5bf-b8ab-4a67-990d-c3a76e2ab353
 ---
 # The 'AddressOf' expression has no effect in this context because the method argument to 'AddressOf' requires a relaxed conversion to the delegate type of the event
-  
-## To correct this error  
-  
-1. Assign the 'AddressOf' expression to a variable, and use the variable to add or remove the method as the handler.  
-  
+
+This warning occurs when you use `AddressOf` with a method in an event handler context where the method signature doesn't exactly match the event's delegate signature. Visual Basic allows "relaxed delegate conversion" in these scenarios, but the `AddressOf` operator becomes redundant because the compiler must create a wrapper delegate anyway.
+
+## Understand relaxed delegate conversion
+
+Visual Basic allows you to assign methods to delegates even when their signatures don't match exactly, as long as the conversion is safe. This is called "relaxed delegate conversion" and includes:
+
+- **Widening parameter types**: A method parameter can be a wider type than the delegate parameter.
+- **Widening return types**: A method return type can be narrower than the delegate return type.  
+- **Omitting parameters**: A method can have fewer parameters than the delegate.
+- **Dropping return values**: A function can be assigned to a `Sub` delegate.
+
+## When this warning occurs
+
+This warning appears when Visual Basic can handle the assignment through relaxed conversion, making the explicit `AddressOf` unnecessary. For example:
+
+```vb
+' Event delegate expects (sender As Object, e As EventArgs)
+Public Event SomeEvent As EventHandler
+
+' Method with exact signature - no warning
+Sub Handler1(sender As Object, e As EventArgs)
+    ' Implementation
+End Sub
+
+' Method with omitted parameters - causes BC42328
+Sub Handler2()
+    ' Implementation  
+End Sub
+
+' Using AddHandler with exact match - no warning
+AddHandler SomeEvent, AddressOf Handler1
+
+' Using AddHandler with relaxed conversion - causes BC42328
+AddHandler SomeEvent, AddressOf Handler2
+```
+
+## Why AddressOf appears to have "no effect"
+
+The warning message states that `AddressOf` "has no effect" because:
+
+1. Visual Basic must create a wrapper delegate for the relaxed conversion regardless
+2. The `AddressOf` operator doesn't change how the compiler handles the conversion
+3. The same result occurs whether you use `AddressOf` or not in relaxed conversion scenarios
```

Comment:
I couldn't tell if 1-3 were mutually exclusive, or *all* true. Maybe this is clearer?

```suggestion
The warning message states that `AddressOf` "has no effect" because Visual Basic must create a wrapper delegate for the relaxed conversion regardless. The `AddressOf` operator doesn't change how the compiler handles the conversion: the same result occurs whether you use `AddressOf` or not in relaxed conversion scenarios.
```